### ISSUE TITLE
Use std::string instead of raw c strings

### ISSUE THREAD

**Issue by objarni** [primaryodors/podock#47]
I can see a lot of `char*` being used, with the corresponding allocations/deallocations, instead of automatically managed `std::strings`. The runtime performance is likely better for `std::string`, and this is anecdotal evidence of concatenation being 3x better:

https://stackoverflow.com/questions/21946447/how-much-performance-difference-when-using-string-vs-char-array

Would you mind if I started refactoring to `std::string`?

**Comment by ssepeq** [comment]
By all means!

**Comment by ssepeq** [comment]
I should let you know that chop_spaced_fields() in point.cpp might present a challenge with this. It takes an input line from a file and divides it up into an array based on a separator character, much like the explode() function of PHP or string.split() in Javascript. An extra function of chop_spaced_fields() is if the separator is space, it treats tabs as a synonym of space. If I recall, it is actually turning the separator chars to nulls and returning an array of char* pointing to the fields from the input char*.

Along the way, I wonder if the std::string conversion would happen to fix issue #9?

**Comment by objarni** [comment]
Possibly it will? We'll see! :)

Just pushed first change, it's for Point::printable() return type. We got a some diff's in some report, since cout << shows floats slightly differently than printf %f, but I just approved that (feels insignificant)

**Comment by ssepeq** [comment]
Agreed about the change to floats. Anything smaller than about a hundredth of an angstrom is trivial to molecular structure, and it looks like we have a thousand times that precision. Awesome!
